src/Sylius/Bundle/CoreBundle/spec/Checkout/InvoiceNumberGeneratorBasedOnIdsSpec.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Bundle\CoreBundle\Checkout;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Order\Model\OrderInterface;
+use Sylius\Component\Payment\Model\PaymentInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class InvoiceNumberGeneratorBasedOnIdsSpec extends ObjectBehavior
+{
+    public function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Bundle\CoreBundle\Checkout\InvoiceNumberGeneratorBasedOnIds');
+    }
+
+    public function it_is_invoice_number_generator()
+    {
+        $this->shouldImplement('Sylius\Bundle\CoreBundle\Checkout\InvoiceNumberGeneratorInterface');
+    }
+
+    public function it_generate_random_invoice_number(OrderInterface $order, PaymentInterface $payment)
```

Comment:
Not really that random. It's tempting and I did it few times too, but avoid copying specs, you loose the best part of speccing. ;)